### DISCUSSION TITLE
Migration that performs data transformation

### DISCUSSION THREAD

**Discussion by brcarey** [prisma/prisma#6959] [Q&A] [answered]
Hi,

Wondering what the recommended approach is for data migrations that also need to transform some data? 

Cheers

**Comment by rozsival** [comment]
I'm having the same question right now. I need to do a large migration that transforms data from one table to another with a different shape. Writing such transformation in pure SQL is incredibly difficult. It would be much better to run some kind of pre and post migration scripts which do the transformation using `PrismaClient`.

**Reply by brcarey** [reply]
Doing it in code would be good (I haven't checked as to if the migrations API is extensible) - but I can handle doing it in sql, if the guidance is just to modify the generated sql and add migration code (you're right, much harder work)

**Comment by brcarey** [comment, marked as answer]
Turns out there is some guidance around this, which involves using the `--create-only` CLI arg and then modifying the generated SQL to suit your migration needs - this will do for me (although I do agree that a hook to run some js code before/after migrations could be useful)

See docs: https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/customizing-migrations

**Reply by brcarey** [reply]
FYI @rozsival

**Reply by rozsival** [reply]
Thanks @brcarey, I found out about that option too. The thing is, it does not suit my needs at the moment. The migration I need to execute is really large and writing the data transform in pure SQL is a no-go. I'm trying to come up with some other solution. Hopefully, Prisma Migrate will support migration scripts in the future.

**Reply by Jolg42** [reply]
We have an issue for a feature that could be interesting for you to follow here [Imperative Migrations with a TypeScript DSL #4688](https://github.com/prisma/prisma/issues/4688)

**Reply by brcarey** [reply]
This looks fantastic and exactly what I am after - is there an avenue to contribute to this feature?

**Reply by Jolg42** [reply]
Thanks for asking! ❤️ We want to first deeply think about it and do the Product work, so now I would say it's not the right time to try implementing something that is not completely designed yet 😃 Your excitement is appreciated though!

**Reply by brcarey** [reply]
No worries, will keep an eye on the issue - would really bring prisma inline with other db tools like alembic, great work
